assembly/JSON.ts

```diff
@@ -183,24 +183,30 @@ export abstract class Value {
 }
 
 export class Str extends Value {
+  private escape_map:Map<string, string> = new Map<string, string>();
```

Comment:
This means an extra allocation for each string instance.  You could make it a static field, but I think the better route here is to make a function with a big switch statement. E.g.

```ts
function escapeChar(char: string): i32 {
   switch (char) {
   case 0x22: return '\\"';
   ...
   default: return char;
}

```